extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php

```diff
@@ -30,11 +30,10 @@ private function getCategoryRequestUrl() {
 	}
 
 	/**
-	 * Get category names for all requested categoryIds
-	 * @param $categoryIds requested category ids
-	 * @return array with name and id pairs for all valid requested categories
+	 * Get raw api response for all categories
+	 * @return raw api response with category names and ids
 	 */
-	public function getCategoryNames( $categoryIds ) {
+	private function getRawCategories() {
 		$memcKey = wfMemcKey( __METHOD__, self::MCACHE_VER );
 		$rawData = WikiaDataAccess::cache(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$rawData". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)